### ISSUE TITLE
Checkbox Column : Checkboxes are not selected with select_all after pjax reload

### ISSUE THREAD

**Issue by devsuitup** [kartik-v/yii2-grid#477]
Hi,

I noticed that when a grid have been  reloaded with pajax, then the select all selects the lines, and doesn't check the boxes ; then, getSelectedRows returns an empty array.

I don't think that was the case a while ago. Did anything changed lately ? 

Is there a fix ?

I couldn't identify where this could come from ; the select_all function from gridview applied a .prop() on the checkboxes, I don't understand why they are not checked and selectable as such by getSelectedRows.

Thanks!

**Comment by fmp-martins** [comment]
I have the precise same issue.
I was able to perform a work around for it:

```
$pjaxRegister = "
$(document).on('pjax:complete', function() { 
	$(\"#gridid\").yiiGridView(\"setSelectionColumn\", {\"name\":\"selection[]\",\"multiple\":true,\"checkAll\":\"selection_all\"});
})";
$this->registerJs($pjaxRegister, \yii\web\View::POS_END);
```

The problem is that when I do it, the checkboxes do work but the datepicker (a \kartik\daterange\DateRangePicker component) stops working, so, not a solution at all.
At this point I get a javascript "**TypeError: settings is undefined yii.grid...93:13**" error.

I also found something interesting. While trying to solve the issue, I left some bad javascript in that pjax:complete script that had a syntax error, which caused some secondary effect (I think stopping the execution of other kartik pjax:complete code) and it started to work all! :D

So, there was a bug that made my application work! :D
Funny, but hardly a solution.

**Comment by devsuitup** [comment]
My work around is this one : 
`
$(document).on('click', '[name="selection_all"]', function() {
    $("[name='selection[]']").prop('checked', $(this).prop('checked')).trigger('change');
});
`

Not the best, though it works...!

**Comment by markmarco16** [comment]
I have the same error, after change page or toggle all, check all return empty.

**Comment by kartik-v** [comment]
Duplicate and Resolved via #511.
